lib/rest-client.js

```diff
@@ -19,13 +19,17 @@ module.exports = CoreObject.extend({
     });
   },
 
-  upload: function(keyPrefix, revisionKey, value) {
+  upload: function(keyPrefix, revisionKey, revisionData, value) {
     var uploadKey = keyPrefix + ':' + revisionKey;
     var requestBody = {
       id: uploadKey,
       body: value
     };
 
+    if(revisionData) {
+      requestBody.revision_data = JSON.stringify(revisionData);
```

Comment:
I figured it may be easier for API authors to just store the data without having to care about what it contains, but as of now it turns out my backend _does_ care what the data contains (in order to display a dashboard) so it's probably for the best. I'll update this PR to pass an object instead.